mllib/src/main/scala/org/apache/spark/ml/feature/Word2Vec.scala

```diff
@@ -194,16 +197,19 @@ class Word2VecModel private[ml] (
    */
   override def transform(dataset: DataFrame): DataFrame = {
     transformSchema(dataset.schema, logging = true)
-    val bWordVectors = dataset.sqlContext.sparkContext.broadcast(wordVectors)
+    bcWordVectors match {
+      case None => bcWordVectors = Some(dataset.sqlContext.sparkContext.broadcast(wordVectors))
+      case _ =>
+    }
     val word2Vec = udf { sentence: Seq[String] =>
```

Comment:
Create a local variable here (e.g. `val localBcWordVectors = bcWordVectors`) and use on L209. The current code expands to `val model = this.bcWordVectors.get...` which means that closure capture will serialize `this` (the current object) with the task.